rust/rope/src/diff.rs

```diff
@@ -245,21 +276,85 @@ impl DiffBuilder {
     }
 }
 
+/// An interval for a specific rope.
+#[derive(Debug, Clone)]
+pub struct RopeSlice<'a> {
+    rope: &'a Rope,
+    range: Range<usize>,
+}
+
+impl PartialEq for RopeSlice<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        let mut it1 =
+            self.rope.iter_chunks(Interval::from(self.range.clone())).flat_map(|x| x.chars());
+        let mut it2 =
+            other.rope.iter_chunks(Interval::from(other.range.clone())).flat_map(|x| x.chars());
+
+        let mut looping = true;
+
+        while looping {
+            if let Some(c1) = it1.next() {
+                if let Some(c2) = it2.next() {
+                    looping = c1 == c2;
+                } else {
+                    return false;
+                }
+            } else {
+                return it2.next().is_none();
+            }
+        }
+
+        false
```

Comment:
This way it is possible to avoid resetting `looping` variable.
```rust
loop {
    let c1 = it1.next();
    let c2 = it2.next();

    if c1 != c2 {
        return false;
    }

    if c1.is_none() {
        return true;
    }
}
```

Reply:
Fixed.